tutorials/evals_ergonomics.ipynb

```diff
@@ -0,0 +1,205 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Evals Ergonomics"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We aim to support evals for three different cases:\n",
+    "\n",
+    "1. Running evals on an exported pandas dataframe and importing back into Phoenix,\n",
+    "2. Running evals during LLM application execution via our callback system,\n",
+    "3. Running evals post-hoc.\n",
+    "\n",
+    "Note: This notebook uses `phoenix.evals` everywhere for convenience, but we'll continue to keep `evals` in `experimental` for the time being."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Running evals on an exported pandas dataframe and importing back into Phoenix\n",
+    "\n",
+    "The user is actively experimenting with their trace data in the form of a pandas dataframe. They wish to compute some evals, perhaps using our custom evaluators or perhaps using their own bespoke code, upload their evaluations, and see their evaluations reflected in Phoenix."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import phoenix as px\n",
+    "from phoenix.evals import ClassificationPromptTemplate, LLMClassifier\n",
+    "from phoenix.experimental.callbacks.langchain_tracer import OpenInferenceTracer\n",
+    "\n",
+    "session = px.launch_app()\n",
+    "trace_df = session.export()\n",
+    "tracer = OpenInferenceTracer()\n",
+    "# Run your LLM application...\n",
+    "\n",
+    "trace_df = session.export_dataframe(\"span_kind == 'retriever'\")\n",
+    "# massage the data...\n",
+    "\n",
+    "# compute relevance with a new prompt template\n",
+    "model = OpenAIModel(model_name=\"gpt-4\")\n",
+    "prompt_template = ClassificationPromptTemplate(\n",
+    "    template=\"some prompt template the user is testing out with {context} and {question}\",\n",
+    "    rails=[\"relevant\", \"irrelevant\"],\n",
+    ")\n",
+    "clf = LLMClassifier(model=model, prompt_template=prompt_template)\n",
+    "trace_df[\"relevant\"] = clf.predict_dataframe(trace_df)\n",
+    "\n",
+    "# import back into phoenix\n",
+    "# accepts pandas dataframe or pandas series indexed with the same span IDs\n",
+    "session.import_evals(trace_df[\"relevant\"])"
```

Comment:
this is probably too many lines and too much manual steps - evals will have to be runnable on spans but the user shouldn't be forced to think about dataframes in this way. All they care about is the evals.

In addition we can't think of evals as singular things - it's really the derived metrics that will matter

Reply:
Overhauled this section. Simplified and created a function for running multiple evals at once.